avm/res/cache/redis/tests/e2e/passive-geo-replication/main.test.bicep

```diff
@@ -0,0 +1,157 @@
+targetScope = 'subscription'
+
+metadata name = 'Passive Geo-Replicated Redis Cache'
+metadata description = 'This instance deploys the module with geo-replication enabled.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-cache.redis-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'crpgeo'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies1 'dependencies1.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies1'
+  params: {
+    location: resourceLocation
+    managedIdentityName: 'dep-${namePrefix}-msi-ds-${serviceShort}'
+    pairedRegionScriptName: 'dep-${namePrefix}-ds-${serviceShort}'
+  }
+}
+
+module nestedDependencies2 'dependencies2.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies2'
+  params: {
+    managedIdentityName: 'dep-${namePrefix}-msi-sec-${serviceShort}'
+    location: nestedDependencies1.outputs.pairedRegionName
```

Comment:
I take it Bicep doesn't allow you to directly reference the deploymentScript output directly due to some 'xzy value must be available at compile time' error?

Reply:
I borrowed this pattern from the db-for-mysql/flexibleServer module and didn't try changing the access patterns at all.

Reply:
Fair,. I still think it may add unnecessary complexity - but feel free to resolve if you want to keep it as is. Just mentioning it as I like to keep things simple. 😏